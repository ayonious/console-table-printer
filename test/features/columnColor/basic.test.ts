import { Table } from '../../../index';
import { getTableBody, getTableHeader } from '../../testUtils/getRawData';

describe('Column Color Tests', () => {
  it('should handle basic column colors', () => {
    const p = new Table()
      .addColumn({ name: 'red', color: 'red' })
      .addColumn({ name: 'green', color: 'green' })
      .addColumn({ name: 'blue', color: 'blue' });

    p.addRows([
      { red: 'Red text', green: 'Green text', blue: 'Blue text' },
      { red: '123', green: '456', blue: '789' },
    ]);

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should handle column colors with different alignments', () => {
    const p = new Table()
      .addColumn({ name: 'left_red', color: 'red', alignment: 'left' })
      .addColumn({ name: 'center_green', color: 'green', alignment: 'center' })
      .addColumn({ name: 'right_blue', color: 'blue', alignment: 'right' });

    p.addRows([
      { left_red: 'Left', center_green: 'Center', right_blue: 'Right' },
      { left_red: '111', center_green: '222', right_blue: '333' },
    ]);

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should handle column colors with row colors', () => {
    const p = new Table()
      .addColumn({ name: 'col1', color: 'red' })
      .addColumn({ name: 'col2', color: 'blue' });

    p.addRow({ col1: 'Red col', col2: 'Blue col' }, { color: 'green' });
    p.addRow({ col1: 'Red col', col2: 'Blue col' }, { color: 'yellow' });

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  it('should handle all available colors', () => {
    const p = new Table({
      columns: [
        { name: 'color_name', alignment: 'left' },
        { name: 'sample', alignment: 'left' },
      ],
    });

    // Test all available colors
    const colors = [
      'red',
      'green',
      'yellow',
      'blue',
      'magenta',
      'cyan',
      'white',
      'crimson',
      'white_bold',
      'gray',
    ];

    colors.forEach((color) => {
      p.addRow(
        {
          color_name: color,
          sample: 'Sample Text',
        },
        { color }
      );
    });

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });

  // Original comprehensive test renamed for clarity
  it('should handle complex color combinations with mixed alignments', () => {
    const p = new Table({
      columns: [
        { name: 'red_left_align_index', alignment: 'left', color: 'red' },
        { name: 'right_align_text', alignment: 'right' },
        { name: 'green_value', color: 'green' },
      ],
    });

    p.addRow(
      {
        red_left_align_index: 2,
        right_align_text: 'This row is blue',
        green_value: 10.212,
      },
      { color: 'blue' }
    );
    p.addRow(
      {
        red_left_align_index: 3,
        right_align_text: 'I would like some red wine please',
        green_value: 10.212,
      },
      { color: 'red' }
    );
    p.addRow(
      {
        red_left_align_index: 4,
        right_align_text: 'I would like some cyan wine please',
        green_value: 10.212,
      },
      { color: 'cyan' }
    );
    p.addRow(
      {
        red_left_align_index: 5,
        right_align_text: 'I would like some white_bold wine please',
        green_value: 10.212,
      },
      { color: 'white_bold' }
    );
    p.addRow(
      {
        red_left_align_index: 7,
        right_align_text: 'I would like some green gemuse please',
        green_value: 20.0,
      },
      { color: 'green' }
    );
    p.addRow(
      {
        red_left_align_index: 8,
        right_align_text: 'I would like some gelb bananen bitte',
        green_value: 100,
      },
      { color: 'yellow' }
    );

    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
