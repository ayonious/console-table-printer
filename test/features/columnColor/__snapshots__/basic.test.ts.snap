// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Color Tests should handle all available colors 1`] = `
"┌────────────┬─────────────┐
│[37m [0m[01mcolor_name[0m │[37m [0m[01msample     [0m │
├────────────┼─────────────┤
│[37m [0m[31mred       [0m │[37m [0m[31mSample Text[0m │
│[37m [0m[32mgreen     [0m │[37m [0m[32mSample Text[0m │
│[37m [0m[33myellow    [0m │[37m [0m[33mSample Text[0m │
│[37m [0m[34mblue      [0m │[37m [0m[34mSample Text[0m │
│[37m [0m[35mmagenta   [0m │[37m [0m[35mSample Text[0m │
│[37m [0m[36mcyan      [0m │[37m [0m[36mSample Text[0m │
│[37m [0m[37mwhite     [0m │[37m [0m[37mSample Text[0m │
│[37m [0mcrimson    │[37m [0mSample Text │
│[37m [0m[01mwhite_bold[0m │[37m [0m[01mSample Text[0m │
│[37m [0mgray       │[37m [0mSample Text │
└────────────┴─────────────┘"
`;

exports[`Column Color Tests should handle basic column colors 1`] = `
"┌──────────┬────────────┬───────────┐
│[37m [0m[01m     red[0m │[37m [0m[01m     green[0m │[37m [0m[01m     blue[0m │
├──────────┼────────────┼───────────┤
│[37m [0m[31mRed text[0m │[37m [0m[32mGreen text[0m │[37m [0m[34mBlue text[0m │
│[37m [0m[31m     123[0m │[37m [0m[32m       456[0m │[37m [0m[34m      789[0m │
└──────────┴────────────┴───────────┘"
`;

exports[`Column Color Tests should handle column colors with different alignments 1`] = `
"┌──────────┬──────────────┬────────────┐
│[37m [0m[01mleft_red[0m │[37m [0m[01mcenter_green[0m │[37m [0m[01mright_blue[0m │
├──────────┼──────────────┼────────────┤
│[37m [0m[31mLeft    [0m │[37m [0m[32m   Center   [0m │[37m [0m[34m     Right[0m │
│[37m [0m[31m111     [0m │[37m [0m[32m    222     [0m │[37m [0m[34m       333[0m │
└──────────┴──────────────┴────────────┘"
`;

exports[`Column Color Tests should handle column colors with row colors 1`] = `
"┌─────────┬──────────┐
│[37m [0m[01m   col1[0m │[37m [0m[01m    col2[0m │
├─────────┼──────────┤
│[37m [0m[31mRed col[0m │[37m [0m[34mBlue col[0m │
│[37m [0m[31mRed col[0m │[37m [0m[34mBlue col[0m │
└─────────┴──────────┘"
`;

exports[`Column Color Tests should handle complex color combinations with mixed alignments 1`] = `
"┌──────────────────────┬──────────────────────────────────────────┬─────────────┐
│[37m [0m[01mred_left_align_index[0m │[37m [0m[01m                        right_align_text[0m │[37m [0m[01mgreen_value[0m │
├──────────────────────┼──────────────────────────────────────────┼─────────────┤
│[37m [0m[31m2                   [0m │[37m [0m[34m                        This row is blue[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m3                   [0m │[37m [0m[31m       I would like some red wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m4                   [0m │[37m [0m[36m      I would like some cyan wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m5                   [0m │[37m [0m[01mI would like some white_bold wine please[0m │[37m [0m[32m     10.212[0m │
│[37m [0m[31m7                   [0m │[37m [0m[32m   I would like some green gemuse please[0m │[37m [0m[32m         20[0m │
│[37m [0m[31m8                   [0m │[37m [0m[33m    I would like some gelb bananen bitte[0m │[37m [0m[32m        100[0m │
└──────────────────────┴──────────────────────────────────────────┴─────────────┘"
`;
